core/embed/rust/src/ui/model_mercury/component/number_input_slider.rs

```diff
@@ -72,17 +101,11 @@ impl Component for NumberInputSliderDialog {
 
     fn render<'s>(&'s self, target: &mut impl Renderer<'s>) {
         self.input.render(target);
-
-        let mut str = ShortString::new();
-        let val_pct = (100 * (self.val - self.min)) / (self.max - self.min);
-
-        unwrap!(ufmt::uwrite!(str, "{} %", val_pct));
-
-        shape::Text::new(self.text_area.center(), &str)
-            .with_font(theme::TEXT_NORMAL.text_font)
-            .with_fg(theme::TEXT_NORMAL.text_color)
-            .with_align(Alignment::Center)
-            .render(target);
+        if self.val == self.init_val || self.input.inner().touching {
+            self.init_footer.render(target);
+        } else {
+            self.modified_footer.render(target);
```

Comment:
Have you considered creating just one `footer` and using `self.footer.update_instruction(...)` and `self.footer.update_description(...)` instead of creating two `Footers` and switching their rendering?

Reply:
> Have you considered creating just one `footer` and using `self.footer.update_instruction(...)` and `self.footer.update_description(...)` instead of creating two `Footers` and switching their rendering?

Much better, indeed!

https://github.com/trezor/trezor-firmware/pull/3991/commits/d2c19add05de1c5d68f7bc28c876dbdeac087553#diff-6e51ef35492aeba10734987923ac867ce64c62e70b23c8a78e47ee86c122c4b6R35